cs/src/core/Index/FASTER/FASTERImpl.cs

```diff
@@ -1401,34 +1400,35 @@ internal enum LatchOperation : byte
                     }
                 }
 
+                int actualSize, allocatedSize;
                 if ((request.logicalAddress < hlog.BeginAddress) || (hlog.GetInfoFromBytePointer(request.record.GetValidPointer()).Tombstone))
                 {
-                    recordSize = hlog.GetInitialRecordSize(ref key, ref pendingContext.input.Get(), fasterSession);
+                    (actualSize, allocatedSize) = hlog.GetInitialRecordSize(ref key, ref pendingContext.input.Get(), fasterSession);
                 }
                 else
                 {
                     physicalAddress = (long)request.record.GetValidPointer();
-                    recordSize = hlog.GetRecordSize(physicalAddress, ref pendingContext.input.Get(), fasterSession);
+                    (actualSize, allocatedSize) = hlog.GetRecordSize(physicalAddress, ref pendingContext.input.Get(), fasterSession);
                 }
-                BlockAllocate(recordSize, out long newLogicalAddress, sessionCtx, fasterSession);
+                BlockAllocate(actualSize, out long newLogicalAddress, sessionCtx, fasterSession);
```

Comment:
Shouldn't this be `allocatedSize` (it seems to be aligned to record size, thus possibly bigger than `actualSize`, I haven't checked does `BlockAllocate` align the size to record)?

Reply:
Great catch, yes!

Reply:
We need to allocate to 8-byte-aligned size so that log full scan can reasonably skip "zero byte" segments of the log in 8-byte increments.